utils/treesitter/grammar.js

```diff
@@ -401,13 +401,12 @@ module.exports = grammar({
     deduced_param: ($) =>
       choice(
         $.generic_binding,
-        seq(optional('addr'), $.ident, ':', $._expression)
+        seq($.ident, ':', $._expression),
+        seq('addr', 'self', ':', $._expression)
```

Comment:
I think at the moment we only allow generic bindings or `self`, so this could just be:

```suggestion
        seq(optional('addr'), 'self', ':', $._expression)
```